docs/example_nodejs.md

```diff
@@ -1,34 +1,314 @@
+status and logs. Your terminal will also turn into a status box if you'd like to
+watch your server come up there.
+
+When the server is ready, you will see the status icon turn green. The log pane will display:
+> Server running at http://localhost:8000/
+
+<figure>
+  <a class="is-image" title="The server is up!" href="https://cloud.tilt.dev/snapshot/todo">
+    <img src="assets/docimg/example-nodejs-serverup.png">
+  </a>
+  <figcaption>The server is up! (Click the screenshot to see an interactive view.)</figcaption>
+</figure>
+
+## Step 1: Let's Add Benchmark Trickery
+
+Before we try to make this faster, let's measure it.
+
+In addition to running things in your cluster, Tilt can run commands locally.
+You can direct Tilt to execute existing scripts or arbitrary shell commands on your own machine.
+
+We want to measure the time from code change to new process. To do that, we'll
+use [`local_resource`](local_resource.html), which lets you locally
+run scripts, shell code, or servers, and manage them from your sidebar like any
+other Tilt resource.
+
+First, we add a `local_resource` to our
+[Tiltfile](https://github.com/windmilleng/tilt-example-nodejs/blob/master/1-measured/Tiltfile)
+that records the current time, then kicks off an update.
+
+```python
+# Records the current time, then kicks off a server update.
+# Normally, you would let Tilt do deploys automatically, but this
+# shows you how to set up a custom workflow that measures it.
+local_resource(
+  'deploy',
+  'date +%s > start-time.txt'
+)
 
-We hear you! We're working on a better guide for this.
+k8s_resource('example-nodejs', port_forwards=8000,
+    resource_deps=['deploy']
+)
+```
+
+The `local_resource()` call creates a local resource named `deploy`. The second
+argument is the command that it runs.
+
+We've also modified our server itself to read that start time and display the time elapsed in the HTML it serves:
+```javascript
+const fs = require('fs');
+let timeSince = 'N/A';
+...
+app.get('/', (req, res) => {
+    res.render('index.mustache', {
+        time: timeSince,
+    });
+});
+
+app.listen(8000, () => {
+    timeSince = getSecsSinceDeploy();
+    console.log('Server running at http://localhost:8000/');
+});
+
+function getSecsSinceDeploy() {
+    let curTimeMs = new Date().getTime();
+    let contents = fs.readFileSync('/app/start-time.txt', 'utf8');
+    let startTimeMs  = parseInt(contents.trim()) / 10**6;
+    return ((curTimeMs - startTimeMs) / 10**3).toFixed(2)
+}
+```
+Whenever the app starts up, it calls `getSecsSinceDeploy()`, does the math to figure out the time elapsed since the timestamp in `start-time.txt`, and stores that value in a global variable; the app then templates that value into `index.mustache` so that it shows up in the served webpage. (We've added a new dep for this purpose: the templating engine [mustache-express](https://www.npmjs.com/package/mustache-express).)
```

Comment:
this is a very complicated paragraph, can we simplify it a bit?

Reply:
good clal

Reply:
thanks, this is a big improvement!